pkg/alibabacloud/api/api.go

```diff
@@ -84,6 +84,27 @@ func (c *Client) GetInstances(ctx context.Context, vpcs ipamTypes.VirtualNetwork
 	return instances, nil
 }
 
+// UpdateInstance fetches the latest ENIs of the given instanceID,
+// updates the local instanceMap and return
+func (c *Client) UpdateInstance(ctx context.Context, vpcs ipamTypes.VirtualNetworkMap, subnets ipamTypes.SubnetMap, instanceID string, instances *ipamTypes.InstanceMap) (*ipamTypes.InstanceMap, error) {
```

Comment:
This function seems to be very similar to GetInstances, except rather than creating a new map it populates one that's passed in. Could we simplify this somehow?

Reply:
(imo: neither of these really fit into the Alibaba API interface very well which seems like it should just be an actions interface around the base alibaba API client.  Processing/population maps should maybe just be in a helper function?)